### ISSUE TITLE
text editor hotkeys (paste, undo, etc) do not respect user keyboard layout

### ISSUE THREAD

**Issue by justDeeevin** [marc2332/freya#1348]
I'm a colemak-dh user and I noticed that, in the text editor examples, ctrl+x causes an undo, rather than ctrl-z. This is because the x key in colemak-dh is in the place where z would be in qwerty, and onkeydown uses keycodes (that is, key position rather than the character itself). Notably, typing text into the editors does work as expected, using the correct characters for my keyboard layout.

I don't know if the intent for onkeydown is to report the keycode rather than the character itself, but regardless of whether this necessitates a change in onkeydown's behavior, the text editors' behaviors, or perhaps entirely new functionality for input handling itself, users will certainly expect text input hotkeys to respect their keyboard layouts, so that should be fixed.

**Comment by marc2332** [comment]
Mmmm, I see, I will check this out!

**Comment by marc2332** [comment]
Hey @justDeeevin Can you try with this branch? https://github.com/marc2332/freya/pull/1351 I pushed a fix just now that removes the use of any hardware key. If you can't or don't want, it's fine! Seems to work fine from my own testing.
